src/cider/exec/nextgen/function/CiderStringFunctions.cpp

```diff
@@ -392,3 +395,218 @@ convert_string_to_time(const char* str_ptr, const int32_t str_len, const int32_t
   std::string from_str(str_ptr, str_len);
   return dateTimeParse<kTIME>(from_str, dim);
 }
+
+static const size_t npos = -1;
+extern "C" ALWAYS_INLINE size_t cider_find_str_from_left(const char* str1,
+                                                         size_t str1_len,
+                                                         const char* str2,
+                                                         size_t str2_len,
+                                                         size_t start_pos = npos) {
+  if (str1_len < str2_len) {
+    return npos;
+  }
+  size_t real_start_pos = (start_pos == npos) ? 0 : start_pos;
+  for (size_t i = real_start_pos; i < str1_len - str2_len; ++i) {
+    if (!std::memcmp(str1 + i, str2, str2_len)) {
+      return i;
+    }
+  }
+  return npos;
+}
+
+extern "C" ALWAYS_INLINE size_t cider_find_str_from_right(const char* str1,
+                                                          size_t str1_len,
+                                                          const char* str2,
+                                                          size_t str2_len,
+                                                          size_t start_pos = npos) {
+  if (str1_len < str2_len) {
+    return npos;
+  }
+  size_t real_start_pos = (start_pos == npos) ? 0 : start_pos;
+  for (size_t i = real_start_pos; i >= str2_len; --i) {
+    if (!std::memcmp(str1 + i - str2_len, str2, str2_len)) {
+      return i - str2_len;
+    }
+  }
+  return npos;
+}
+
+extern "C" ALWAYS_INLINE int64_t cider_split(char* string_heap_ptr,
```

Comment:
For different branches in this function, are you referring any standards for handling different scenarios? If so, maybe you can give a link here. Same suggestion to other functions?

Reply:
emmm, I referred HeavyDB implementation(which use std::string). Maybe I should list some examples here?

Reply:
Sure. Comments also can help.